application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,224 @@
+        String reportedMessageID = event.getTarget().getId();
+        String reportedMessageChannel = event.getTarget().getChannel().getId();
+        String reportedAuthorName = event.getTarget().getAuthor().getName();
+        String reportedAuthorAvatarURL = event.getTarget().getAuthor().getAvatarUrl();
+        String reportedAuthorID = event.getTarget().getAuthor().getId();
+
+        TextInput body = TextInput
+            .create(REPORT_REASON_INPUT_ID, "Anonymous report to the moderators",
+                    TextInputStyle.PARAGRAPH)
+            .setPlaceholder("What is wrong with the message, why do you want to report it?")
+            .setRequiredRange(3, 200)
+            .build();
+
+        String reportModalComponentID = generateComponentId(reportedMessage, reportedMessageID,
+                reportedMessageChannel, reportedMessageTimestamp, reportedAuthorName,
+                reportedAuthorAvatarURL, reportedAuthorID);
+        Modal reportModal = Modal.create(reportModalComponentID, "Report this to a moderator")
+            .addActionRow(body)
+            .build();
+
+        event.replyModal(reportModal).queue();
+    }
+
+    private boolean handleIsOnCooldown(MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(event.getUser().getIdLong())) {
+            return false;
+        }
+        event
+            .reply("You can only report a message once per %s minutes."
+                .formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        Optional<TextChannel> modMailAuditLog = handleRequireModMailChannel(event);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
+
+        sendModMessage(event, args, modMailAuditLog.orElseThrow());
+    }
+
+    private Optional<TextChannel> handleRequireModMailChannel(ModalInteractionEvent event) {
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = event.getJDA()
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
```

Comment:
y this weird string

Reply:
whats weird with it? what did u expect?

Reply:
("Sorry, there was an issue sending your report to the moderators. We are " + "investigating.")

How is it not?

Reply:
Its because of commit hook, it reached the limit

Reply:
The string split is totally fine. Note that there is a mini trick with code blocks and `\`
```java
"""
Sorry, there was ... We are \
investigating."""
```

Reply:
But its not an issue. constant strings are concatted during compile time. it is a single string after compilation